integration-cli/docker_cli_swarm_test.go

```diff
@@ -194,3 +194,19 @@ func (s *DockerSwarmSuite) TestSwarmNodeTaskListFilter(c *check.C) {
 	c.Assert(out, checker.Not(checker.Contains), name+".2")
 	c.Assert(out, checker.Not(checker.Contains), name+".3")
 }
+
+// Test case for #25375
+func (s *DockerSwarmSuite) TestSwarmPublishAdd(c *check.C) {
+	d := s.AddDaemon(c, true, true)
+
+	name := "top"
+	out, err := d.Cmd("service", "create", "--name", name, "--label", "x=y", "busybox", "top")
+	c.Assert(err, checker.IsNil)
+	c.Assert(strings.TrimSpace(out), checker.Not(checker.Equals), "")
+
+	out, err = d.Cmd("service", "update", "--publish-add", "80:80", name)
+	c.Assert(err, checker.IsNil)
+
+	out, err = d.Cmd("service", "update", "--publish-add", "80:80", name)
```

Comment:
Add a failing case for `--publish-add 80:80 --publish-add 80:20`. That should error out on the client side.

Also add a case for service created with `--publish 80:80` then update to `--publish-add 80:20`.

The verb "add" implies set logic. We consider the exposed cluster ports a set of external ports. Adding a port with a new target port should update the value.